src/main/java/seedu/duke/command/SearchModuleCommand.java

```diff
@@ -15,51 +15,54 @@
 
 public class SearchModuleCommand extends Command {
     public static final String COMMAND_WORD = "search";
-    public static final String COMMAND_USAGE = "search (/code [MODULE_CODE] | /title [KEYWORD])";
-    public static final String COMMAND_DESCRIPTION = "List out all modules that contains a search term"
-            + System.lineSeparator() + "\t * the search term can either be module code or a keyword in module title.";
+    public static final String COMMAND_USAGE = "search /code [SIMILAR_MODULE_CODE] /title [SIMILAR_MODULE_TITLE] "
+            + "/level <level> /sem <semester>";
+    public static final String COMMAND_DESCRIPTION = "List out all modules that contains matching input search fields"
+            + System.lineSeparator() + "\t * Either the module code or title has to be present.";
 
-    // private String toSearchModuleCode;
-    private Map<String, String> params;
     private String toSearchModuleCode;
     private String toSearchModuleTitle;
     private String toSearchLevel;
     private String toSearchSemester;
 
-    private Logger logger;
-
     public static final String SUBSYSTEM_NAME = "SearchModuleCommand";
 
     public SearchModuleCommand(String input) throws YamomException {
         super(input.split("\\s"));
-        params = Parser.parseParams(input);
-
+        Map<String, String> params = Parser.parseParams(input);
         toSearchModuleCode = params.getOrDefault("code", null);
         toSearchModuleTitle = params.getOrDefault("title", null);
         toSearchLevel = params.getOrDefault("level", null);
         toSearchSemester = params.getOrDefault("sem", null);
 
-        if (isAllNull(toSearchModuleCode, toSearchModuleTitle, toSearchLevel, toSearchSemester)) {
-            throw new YamomException("No search field provided! Should be " + COMMAND_USAGE);
+        // if size of params is 0, means no params are given, throw exception
+        if (params.size() == 0) {
+            throw new YamomException("Please input valid search fields to search for! You can search by module code, "
+                    + "module title, level and semester.\n\nType [help] for assistance!");
```

Comment:
should use System.lineSeparator() for consistency

Reply:
which part of it should I use System.lineSeparator()?